rc/tools/git.kak

```diff
@@ -71,22 +77,27 @@ define-command -params 1.. \
     if [ $kak_token_to_complete -eq 0 ]; then
         printf "add\nrm\nreset\nblame\ncommit\ncheckout\ndiff\nhide-blame\nhide-diff\nlog\nnext-hunk\nprev-hunk\nshow\nshow-branch\nshow-diff\ninit\nstatus\nupdate-diff\ngrep\n"
     else
+        # Try cd-ing into the git work tree if exists to make autocompletion slightly more robust
+        cd "$kak_opt_git_work_tree_top_level" >/dev/null 2>&1
         case "$1" in
             commit) printf -- "--amend\n--no-edit\n--all\n--reset-author\n--fixup\n--squash\n"; git ls-files -m ;;
             add) git ls-files -dmo --exclude-standard ;;
             rm|grep) git ls-files -c ;;
         esac
     fi
   } \
-  git %{ evaluate-commands %sh{
-    cd_bufdir() {
-        dirname_buffer="${kak_buffile%/*}"
-        cd "${dirname_buffer}" 2>/dev/null || {
-            printf 'fail Unable to change the current working directory to: %s\n' "${dirname_buffer}"
-            exit 1
-        }
-    }
-
+  git %{
+  set-option global git_work_tree_top_level %sh{
+    # - try to get the toplevel for the `cwd`
+    # - if failed, try to find a toplevel for a given buffile
+    # - if failed, test the current toplevel variable and if it is a functional git repository
+    # - if failed, fallback to `cwd` ()
+    git rev-parse --show-toplevel 2>/dev/null || \
+    ( cd "${kak_buffile%/*}" && git rev-parse --show-toplevel ) 2>/dev/null || \
```

Comment:
Technically we don't want to try this for scratch buffers.. but I don't know how to distinguish scratch buffers here. I guess it's okay to ignore this for now

Reply:
I don't know how to do it as well. I switched to `-C` but the problem remains if some user names their directories like '*git*' or just happens to have scratch buffers named like some directories..? There's probably a bunch of unpleasant corner-cases that might happen.

Speaking of wildcards, I noticed that parameterless `git add/rm` defaults to current buffile which then goes straight to the `git` command which seems to trigger wildcard expansion of the shell in case of scratch buffers like `*git*`.